assets/src/apps/bibliography/citation_model.ts

```diff
@@ -0,0 +1,231 @@
+  'original-publisher'?: string;
+  'original-publisher-place'?: string;
+  'original-title'?: string;
+  page?: string;
+  'page-first'?: string | number;
+  part?: string | number;
+  'part-title'?: string;
+  pmcid?: string;
+  pmid?: string;
+  printing?: string | number;
+  publisher?: string;
+  'publisher-place'?: string;
+  references?: string;
+  'reviewed-genre'?: string;
+  'reviewed-title'?: string;
+  scale?: string;
+  section?: string;
+  source?: string;
+  status?: string;
+  supplement?: string | number;
+  title?: string;
+  'title-short'?: string;
+  url?: string;
+  version?: string;
+  volume?: string;
+  'volume-title'?: string;
+  'volume-title-short'?: string;
+  'year-suffix'?: string;
+  custom?: any;
+}
+
+export const fromEntryType = (entryType: string): CitationModel => {
+  let model: CitationModel = {
+    id: 'temp_id_' + guid(),
+    type: entryType,
+    author: [{ given: '', family: '' }],
+    title: '',
+    issued: { 'date-parts': [[new Date().getFullYear()]] },
+    publisher: '',
+    version: '',
+    shortTitle: '',
+    doi: '',
+  };
+
+  switch (entryType) {
+    case 'article':
+    case 'article-journal':
+    case 'article-newspaper':
+      model = {
+        id: 'temp_id_' + guid(),
+        type: entryType,
+        author: [{ given: '', family: '' }],
+        title: '',
+        'container-title': '',
+        issued: { 'date-parts': [[new Date().getFullYear()]] },
+        volume: '',
+        number: '',
+        page: '',
+        note: '',
+        doi: '',
+        issn: '',
+      };
+      break;
+    case 'book':
+      model = {
+        id: 'temp_id_' + guid(),
+        type: entryType,
+        author: [{ given: '', family: '' }],
+        editor: [{ given: '', family: '' }],
+        title: '',
+        publisher: '',
+        issued: { 'date-parts': [[new Date().getFullYear()]] },
+        volume: '',
+        number: '',
+        page: '',
+        note: '',
+        doi: '',
+        issn: '',
+        isbn: '',
+        url: '',
+      };
+      break;
+    default:
+      break;
+  }
+  return model;
+};
+
+export function isNameField(val: string): boolean {
+  if (
+    val === 'author' ||
+    val === 'chair' ||
+    val === 'collection-editor' ||
+    val === 'compiler' ||
+    val === 'composer' ||
+    val === 'container-author' ||
+    val === 'contributor' ||
+    val === 'curator' ||
+    val === 'director' ||
+    val === 'editor' ||
+    val === 'editorial-director' ||
+    val === 'executive-producer' ||
+    val === 'guest' ||
+    val === 'host' ||
+    val === 'interviewer' ||
+    val === 'illustrator' ||
+    val === 'narrator' ||
+    val === 'organizer' ||
+    val === 'original-author' ||
+    val === 'performer' ||
+    val === 'producer' ||
+    val === 'recipient' ||
+    val === 'reviewed-author' ||
+    val === 'script-writer' ||
+    val === 'series-creator' ||
+    val === 'translator'
+  ) {
+    return true;
```

Comment:
you can just return the condition, no need to specify true and false

Reply:
Suggestion applied